### ISSUE TITLE
How to fix TRANSMISSION 2.92-12 on fresh DSM6 installation DSM 6.0.2-8451 Update 9

### ISSUE THREAD

**Issue by lobocobra** [SynoCommunity/spksrc#2645]
_For new Package Requests, see the guidelines_  

### Setup
_Package Name:_ transmission
_Package Version:_ 2.92-12

_NAS Model:_ DS415+
_NAS Architecture:_
_DSM version:_6.0.2-8451 Update 9

### Expected behavior
Tell us what should happen
If you install transmission from the package manager, the package transmission will install but not run.

### Actual behavior
Tell us what happens instead
nothing happens, there is no message and the daemon does not run

### Steps to reproduce
_1._ install the package transmission in the package manager
_2._ try to run it
_3._ it will not work

#### Package log
no log is created

#### WORK AROUND, how to make it work
1) #install missing packages
    #manually install phyton package from package center
2) #manually install a user called "transmission" and give him access to your transmission download directory
3) #manually change ownership of transmission package directory 
   chown /volume2/@appstore/transmission -R
4) # manually correct change start/stop script of transmission
    vi /var/packages/transmission/scripts/start-stop-status
    #change the command in the "start" section as follows:
    start_daemon ()
    {
     su - ${USER} -c "PATH=${PATH} ; ${TRANSMISSION} -g ${INSTALL_DIR}/var -x ${PID_FILE}"
    }
    # see the added semi-colon which prevents, that "-g" is seen as a command for su (undocumented group selection)

#### fixed ######
=> I did not know where to post, but this fixes the problem, please point me to the developper page, so I can send the fix or report it yourself.

**Comment by Dr-Bean** [comment]
Thanks for reporting, but as you can imagine, we already knew about this :) 
A complete implementation/fix for all affected packages is currently being developed in the `dsm6` branch, as part of #2216. Subscribe to that issue to keep informed of progress.
In the meantime, I'm going to close this issue.

**Comment by zmwangx** [comment]
I also needed to install Transmission on my DSM 6.0.2 box just now, and want to share my experience in case someone stumbles upon this in the future.

After reading #2216, I got the impression that it might be a bad idea to use `synouser` to create `transmission` the usual way (it seems that the shell would be reset to `/sbin/nologin` upon reboot).

Since the issue ultimately boils down to permission denied when trying to execute

    su - transmission -c '/usr/local/transmission/bin/transmission-daemon -g /usr/local/transmission/var -x /usr/local/transmission/var/transmission.pid

as root due to missing entry for `transmission` in `/etc/shadow`, I simply added

    transmission:*:10933:0:99999:7:::

to `/etc/shadow`.

This may not be the best idea, but it does work for me (I haven't rebooted so I can't reliably say if this setup would survive a reboot, but according to my understanding of #2216 it should).

You don't need to touch anything else (OP's modifications are useless anyway; the command is quoted, there's no chance of misinterpretation).

---

**TL;DR**: Add

     transmission:*:10933:0:99999:7:::

to `/etc/shadow`. (Disclaimer: I'm not responsible for any damage to your box.)
